.github/workflows/ci.yml

```diff
@@ -712,3 +725,166 @@ jobs:
+      release_notes: ${{ steps.determine-if.outputs.release_notes }}
+      latest_tag: ${{ steps.determine-if.outputs.latest_tag }}
+      next_tag: ${{ steps.determine-if.outputs.next_tag }}
+    steps:
+
+      - name: Determine if should patch release
+        id: determine-if
+        run: |
+          DO_RELEASE=1
+
+          # Must be on a minor branch to do a patch release
+          if ! [[ $GITHUB_REF_NAME =~ ^[0-9]+\.[0.9]+$ ]]; then
+            echo "Not patch releasing because not on a minor branch"
+            DO_RELEASE=0
+          fi
+
+          # Validate that this commit is that latest commit for the branch using GitHub API
+          # We need to check this in case re-rerunning an old job and there have been new commits since
+          if [[ $DO_RELEASE == "1" ]]; then
+            # https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28
+            AMP='&'
+            RESP_CODE=$(curl -w %{http_code} -s -o __response.json \
+            -X GET https://api.github.com/repos/${GITHUB_REPOSITORY}/commits?sha=${GITHUB_REF_NAME}${AMP}per_page=1 \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ github.token }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            )
+            if [[ $RESP_CODE != "200" ]]; then
+              echo "Unable to read list of commits - HTTP response code was $RESP_CODE"
+              exit 1
+            fi
+            LATEST_REMOTE_SHA=$(jq -r '.[0].sha' __response.json)
+            echo "LATEST_REMOTE_SHA is $LATEST_REMOTE_SHA"
+            echo "LATEST_LOCAL_SHA is $LATEST_LOCAL_SHA"
+            if [[ $LATEST_REMOTE_SHA != $LATEST_LOCAL_SHA ]]; then
+              echo "Not patch releasing because latest remote sha is not equal to latest local sha"
+              DO_RELEASE=0
+            fi
+            # Also validate the sha matches GITHUB_SHA, which is what gha-tag-release will use
+            if [[ $GITHUB_SHA != $LATEST_LOCAL_SHA ]]; then
+              echo "Not patch releasing because GITHUB_SHA is not equal to latest local sha"
+              DO_RELEASE=0
+            fi
+          fi
+
+          # Check is there is an existing tag on the branch using GitHub API
+          # Note cannot use local `git tag` because actions/checkout by default will not checkout tags
+          # and you need to checkout full history in order to get them
+          LATEST_TAG=""
+          NEXT_TAG=""
+          if [[ $DO_RELEASE == "1" ]]; then
+            # https://docs.github.com/en/rest/git/tags?apiVersion=2022-11-28
+            RESP_CODE=$(curl -w %{http_code} -s -o __response.json \
+            -X GET https://api.github.com/repos/${GITHUB_REPOSITORY}/tags?per_page=100 \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ github.token }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            )
```

Comment:
We can use https://docs.github.com/en/rest/git/refs?apiVersion=2022-11-28#list-matching-references instead

e.g. where `GITHUB_REPOSITORY` is "silverstripe/silverstripe-framework" and the branch being pushed to is "4.13":
https://api.github.com/repos/silverstripe/silverstripe-framework/git/matching-refs/tags/4.13 gives us all tags that start with 4.13! We just need to grab the list item in the list, which simplifies the code below that's trying to get the latest tag.

Reply:
I'd say it'll probably actually complicate the code below due to the need to strip the `refs/tags/` out of the `ref` attribute.  Currently we use
`LATEST_TAG=$(jq -r '.[].name' __response.json | grep -E ^$ESCAPED_REF_NAME\.[0-9]+$ | sort -V -r | head -n 1)`

We'd need to change it to something like

`LATEST_TAG=$(jq -r '.[].name' __response.json | <remove refs/tags/> |grep -E ^[0-9]+\.[0-9]+\.[0-9]+$ | sort -V -r | head -n 1)`

Though I'm not sure how to do the `<remove refs/tags/>` bit in bash at this point in the pipe chain

I would prefer to do it this way just not needing to worry about the 100 tag limitation, though as mentioned offline the [tags API for framework](https://api.github.com/repos/silverstripe/silverstripe-framework/tags?per_page=100) goes all the way back to 4.5.1 which is way out of support, so the 100 limit should be fine in practice

Reply:
We know that we only have _relevant_ tags in this list. We also know that they're in order from either oldest to newest or lowest semver to highest semver (which for our purposes are identical sort orders). So we can just grab the last item in the list without resorting or using grep etc - that would look something like this:
```bash
LATEST_TAG=$(jq -r '.[-1:][].ref' __response.json)
```
We can then do something like `LATEST_TAG=${LATEST_TAG#refs/tags/}` to remove `refs/tags` - but I'm sure there are plenty of other ways to do it once we have it as a variable.

If you do want to keep the grep and sort in, just change your regex to `^ref/tags/[0-9]+\.[0-9]+\.[0-9]+$` and remove the `ref/tags/` part after we've got it in a variable. So:
```bash
LATEST_TAG=$(jq -r '.[].ref' __response.json | grep -E ^refs\/tags\/[0-9]+\.[0-9]+\.[0-9]+$ | sort -V -r | head -n 1)
LATEST_TAG=${LATEST_TAG#refs/tags/}
```

If you want to do everything in one line like you have been, you can use a positive lookbehind to remove the `refs/tags/` part as part of the match, like so:
```bash
LATEST_TAG=$(jq -r '.[].ref' __response.json | grep -Po '(?<=^refs\/tags\/)[0-9]+\.[0-9]+\.[0-9]+$' | sort -V -r | head -n 1)
```

Reply:
I think knowing we _are only_ dealing with tags for the minor release we care about is a big plus, and not having the 100 tag limit is a theoretical win as well. I've given three ways to handle this, ranging from "just grab the last item 'cause we know it's the one we need" to "keep all your existing logic and just change the regex slightly to account for the prefix".

Reply:
Updated to use the positive lookbehind within the pipe chain, works nice cheers